libgnucash/engine/mocks/fake-qofquery.cpp

```diff
@@ -0,0 +1,155 @@
+#include <config.h>
+
+#include <list>
+
+#include "fake-qofquery.h"
+#include "gmock-qofbook.h"
+
+
+
+/* class QofFakeQueryPool */
+
+static class QofFakeQueryPool
```

Comment:
This needs Doxygen documentation explaining how to use the query pool, highlighting the life-cycle of a QofFakeQuery and the importance of ensuring that `query_pool->request_query` and `query_pool->release_query` are called on every `QofFakeQuery` or the test program will `ASSERT`.

Reply:
No, it's not necessary, that a `QofFakeQuery` object is requested (and released) by the test application. It can also be removed from the pool, if it has not been consumed. See the implementation of `QofFakeQueryPool::removeQuery()`. If a `QofFakeQuery` object is not found in the list of consumed query objects, it is searched in the list of new query objects. There are only two assertions in this function:
- when the `QofFakeQuery` object is currently in use (requested, but not released yet)
- when the `QofFakeQuery` object has never been added to the pool (which is actually not possible)

Reply:
There are three asserts: One in `releaseQuery` if the query isn't in use as well as the two in `removeQuery` that you describe. `removeQuery` is run by the query's dtor and if the query is never requested it will indeed happily delete it. If OTOH it has been requested and not released then the `removeQuery` will assert; since there's no point in creating the query if you're not going to request it, it's important to request and release during the query's lifetime and the documentation needs to explain that.

Reply:
What doxygen style do you prefer? I've seen Javadoc style quite often in the code. Is there any style guide for code documentation as well?

Reply:
I think most of it's `/** .. */` with `@` for keywords. Best to stay consistent with that.

Reply:
Class `QofFakeQueryPool` is a `static` class and at the moment doxygen does not generate documentation for static classes according to the current doxygen configuration (see `EXTRACT_STATIC = NO`). That means doxygen comments on this class are not part of the doxygen documentation.

Is it intended, that `EXTRACT_STATIC` is set to `NO`? And should class `QofFakeQueryPool` be documented anyway?

Reply:
That's reasonable, `QofFakeQueryPool` is an implementation detail so it probably shouldn't be documented in Doxygen. The documentation would be better on `QofFakeQuery`.